blob/s3blob/s3blob_test.go

```diff
@@ -47,29 +47,39 @@ const (
 
 type harness struct {
 	session *session.Session
+	opts    *Options
 	rt      http.RoundTripper
 	closer  func()
 }
 
 func newHarness(ctx context.Context, t *testing.T) (drivertest.Harness, error) {
 	sess, rt, done := setup.NewAWSSession(t, region)
-	return &harness{session: sess, rt: rt, closer: done}, nil
+	return &harness{session: sess, opts: nil, rt: rt, closer: done}, nil
+}
+
+func newHarnessUsingLegacyList(ctx context.Context, t *testing.T) (drivertest.Harness, error) {
+	sess, rt, done := setup.NewAWSSession(t, region)
+	return &harness{session: sess, opts: &Options{UseLegacyList: true}, rt: rt, closer: done}, nil
 }
 
 func (h *harness) HTTPClient() *http.Client {
 	return &http.Client{Transport: h.rt}
 }
 
 func (h *harness) MakeDriver(ctx context.Context) (driver.Bucket, error) {
-	return openBucket(ctx, h.session, bucketName, nil)
+	return openBucket(ctx, h.session, bucketName, h.opts)
 }
 
 func (h *harness) Close() {
 	h.closer()
 }
 
 func TestConformance(t *testing.T) {
-	drivertest.RunConformanceTests(t, newHarness, []drivertest.AsTest{verifyContentLanguage{}})
+	drivertest.RunConformanceTests(t, newHarness, []drivertest.AsTest{verifyContentLanguage{usingLegacyList: false}})
+}
+
+func TestConformanceUsingLegacyList(t *testing.T) {
+	drivertest.RunConformanceTests(t, newHarnessUsingLegacyList, []drivertest.AsTest{verifyContentLanguage{usingLegacyList: true}})
```

Comment:
This will run all the conformance tests one more time. It seems a little bit redundant. I don't have any quick suggestions over the top of my head and a lot of the conformance tests do use List in this case.

Reply:
Acknowledged. I don't think there's a lot of cost running them again. If we ended up with lots of combinations of parameters we may reconsider.